extensions/kusto/src/iconProvider.ts

```diff
@@ -0,0 +1,21 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as constants from './constants';
+
+const cloudIcon = 'kusto:cloud';
+
+export class MssqlIconProvider implements azdata.IconProvider {
+	public readonly providerId: string = constants.sqlProviderName;
+	public handle: number;
+	getConnectionIconId(connection: azdata.IConnectionProfile, serverInfo: azdata.ServerInfo): Thenable<string> {
+		let iconName: string = undefined;
```

Comment:
type is string|undefined, not string

Reply:
I recommend turning on strict mode for this extension since its a new extension and far easier to add that change